application_sandbox/async_compute/main.cpp

```diff
@@ -419,35 +464,54 @@ class ASyncThreadRunner {
     }
 
     std::lock_guard<std::mutex> lock(data_mutex_);
-    if (mailbox_buffer_ == -1) {
+    if (mailbox_buffers_.empty()) {
```

Comment:
This isn't really a mailbox anymore. We probably want to change this naming to fifo.

FWIW the original intention here was that the async would run independently from the draw. Meaning if could tick at a rate that was independent from the draw. (If draw took longer you could get double updates of compute, or if compute was slow you would just render what you arleady had).

Reply:
Yeah, I was trying to find a way to preserve this mailbox behaviour as closely as possible, within the constraints of the spec. This does still allow async compute to have multiple ticks per render, but only up to the number of buffers.

I only see 2 ways to make this legal:
1. The gfx queue has to acquire and release each buffer before it can be used again (this PR).
2. The compute queue does not release the buffers it puts in the mailbox. When the render is ready to take a buffer from the mailbox, it would need to submit the release on an async compute queue. For the render to not block until the current async compute completes, I think this would require at least 2 async compute queues.
3. Get rid of the queue family transfers and just use concurrent sharing mode.

Reply:
Right I guess I understand the issue now this was because of bad transfers. Ok I can live with this approach. I suspect concurrent sharing is probably how this would look in practice as it cleans up the interface (at MAYBE the cost of performance but probably not really).

Reply:
lgtm from me for this part then

Reply:
I tried to make things a bit clearer here:
- I restored `mailbox_buffer_` as a single buffer index, behaving properly like a mailbox.
- Separately, there is now a vector `released_buffers_` that contains all the buffers that have been released by async compute.
- I added a helper method `TransferBuffer` to handle the queue submit to acquire and/or release a buffer on the graphics queue.